docs/v1/P4Runtime-Spec.mdk

```diff
@@ -414,6 +414,27 @@ caption: "P4Runtime Reference Architecture." }
 [reference-architecture]: build/reference-architecture.[svg,png] \
 { height: 7cm; page-align: here }
 
+## P4Runtime Service Implementation
+
+The P4Runtime API is implemented by a program that runs a gRPC server which
+binds an implementation of auto-generated P4Runtime Service interface. This
+program is called the "P4Runtime server." The server must listen on TCP port
+9559 by default, which is the port that has been allocated by IANA for the
+P4Runtime service. Servers should allow users to override the default port
+using a configuration file or flag when starting the server. Uses of other
+port numbers as the default should be discontinued.
+
+### Security concerns
+
+Appropriate measures and security best practices must be in place to protect
+the P4Runtime server and client, and the communication channel between the two.
+For example, firewalling and authenticating the incoming connections to the
+P4Runtime server can prevent a malicious actor from taking over the switch.
+Similarly, using TLS to authenticate and encrypt the gRPC channel can prevent
+man-in-the-middle attacks between the server and client. Mutual TLS (mTLS) may
+be used to facilitate the authentication of the client by the server and
+vice-versa.
```

Comment:
Thanks for moving this, but you forgot to remove the original "Security concerns for P4Runtime" section?

Reply:
oops good catch.